### DISCUSSION TITLE
Complete a task manually

### DISCUSSION THREAD

**Discussion by meggarr** [kagkarlsson/db-scheduler#462] [Q&A] [answered]
Hi folks,

Do you know if there's a way to complete the task manually after execution ? E.g. I have an async task executed by another microservice, upon completion, it will send back a completion message, then I want to complete the DB scheduler task manually.

Is this possible ?

Thanks in advance.

**Comment by kagkarlsson** [comment]
Not sure I fully understand. If you other microservice has access to the database, it can delete/cancel the scheduled task manually (as long as it is not currently running). You can instantiate `SchedulerClient`, you do not need a full `Scheduler`

**Reply by meggarr** [reply]
Hi @kagkarlsson , & I meant task instance. E.g. I init a task instance, in which it will kick a remote microservice to do a job, after the job completes, the remote microservice will send a result notification.

I want to know if there is a way to leave the task instance un-completed, and then when the result is notified from remote, can the `SchedulerClient` complete the task instance with a function method ?

**Reply by kagkarlsson** [reply, marked as answer]
Ok. That is not possible, the instance must be completed (deleted or rescheduled) else it will be considered dead. 

You might reschedule the instance a bit into the future though. And the "completion" would be to delete the instance when the microservice job completes.

**Reply by meggarr** [reply]
Thanks!
